### ISSUE TITLE
Converter.py TemplateNotFound

### ISSUE THREAD

**Issue by liaoyuzh** [DahlitzFlorian/markdown-to-html-generator-article-snippets#3]
As the title said, when I run the `converter.py` file, it complains about missing template, which refers to the `layout.html`.
The project is being git cloned, and everything was not moved. Double checked the `layout.html` was there. dont know why this happened. The error occurs with the lin `doc = env.get_template(str(BLOG_TEMPLATE_FILE)).render(`

**Comment by DahlitzFlorian** [comment]
Hello @liaoyuzh, thanks for raising this issue. I have cloned the repo as it is, installed the dependencies from the `requirements.txt` and ran the following command from the project's root directory:
```shell
$ python app/services/convert.py
rendering app/posts/post1.md
rendering app/posts/post2.md
```
I used Python 3.8 for it:
```shell
$ python --version --version
Python 3.8.5 (default, May 27 2021, 13:30:53)
[GCC 9.3.0]
```
From the input, you have provided I assume, that you have run it from the `services` directory. Please correct me if I'm wrong. You need to run the command from the project's root directory, so all files are located properly.

Feel free to reopen the issue and provide further details about your setup and environment if the above statement doesn't help.
Thanks again and have a nice day!

**Comment by liaoyuzh** [comment]
Thanks for taking your time answering my question, I have re-done your steps: 
1) git clone repo
2) `pip install -r requirements.txt`
3) `python app/services/convert.py` which is run at root directory
and I am using Python 3.8.1 which I think has no problem

but I feel necessary to post this detailed error message
```
$ python app/services/convert.py
rendering app\posts\post1.md
Traceback (most recent call last):
  File "app/services/convert.py", line 93, in <module>
    generate_entries()
  File "app/services/convert.py", line 46, in generate_entries
    doc = env.get_template(str(BLOG_TEMPLATE_FILE)).render(
  File "C:\Users\Terre\AppData\Roaming\Python\Python38\site-packages\jinja2\environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "C:\Users\Terre\AppData\Roaming\Python\Python38\site-packages\jinja2\environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Terre\AppData\Roaming\Python\Python38\site-packages\jinja2\loaders.py", line 125, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Terre\AppData\Roaming\Python\Python38\site-packages\jinja2\loaders.py", line 194, in get_source
    pieces = split_template_path(template)
  File "C:\Users\Terre\AppData\Roaming\Python\Python38\site-packages\jinja2\loaders.py", line 35, in split_template_path
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: app\templates\shared\layout.html
```

**Comment by DahlitzFlorian** [comment]
Thanks for sharing the message. Seems to me your operating system is Windows and Python was installed via the Windows Store, is that correct?

**Comment by liaoyuzh** [comment]
> Thanks for sharing the message. Seems to me your operating system is Windows and Python was installed via the Windows Store, is that correct?

not sure how I downloaded that python environment in the first place. However when I setup a conda environment with those dependencies and run the command, this `jinja2.exceptions.TemplateNotFound` still pops up. I don't know how thats related to the error message.

**Comment by DahlitzFlorian** [comment]
@liaoyuzh I was able to reproduce and fix the issue - thanks for reporting! The articles will be adjusted in a couple of hours, so they are compliant with the code in the repository.
The issue is that jinja2 seems not to be able to handle absolute paths (at least not reliably in every environment). Consequently, I had to change the search path supplied to the `FileSystemLoader` and use relative strings to the templates.
